### PR TITLE
Fix a race in pkg/discovery/memory

### PR DESCRIPTION
**Pull request by AkihiroSuda**
**\- What I did**
Fixed a race found in #22964

**\- How I did it**
Introduced a mutex

**\- How to verify it**
`TESTDIRS=pkg/discovery/memory BUILDFLAGS=-race TESTFLAGS="-test.count 1000" make test-unit`

Signed-off-by: Akihiro Suda suda.akihiro@lab.ntt.co.jp

### CODE REVIEW

File: pkg/discovery/memory/memory.go [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
 	"github.com/docker/docker/pkg/discovery"
 )
 
-// Discovery implements a descovery backend that keeps
+// Discovery implements a discovery backend that keeps
 // data in memory.
 type Discovery struct {
 	heartbeat time.Duration
 	values    []string
+	mu        sync.RWMutex
```
Comment (cpuguy83) [inline pkg/discovery/memory/memory.go]: I think we can get away with a regular mutex here.

Comment (AkihiroSuda) [inline pkg/discovery/memory/memory.go]: Done

**Comment by AkihiroSuda** [comment]
The failure in gccgo is unrelated to my change (known flaky test https://github.com/docker/docker/issues/22879)

**Comment by vdemeester** [comment]
LGTM :angel: 
/cc @LK4D4 @runcom @icecrime @cpuguy83

**Comment by cpuguy83** [comment]
LGTM

**Pull request merged**